### DISCUSSION TITLE
React Hook Form | useWatch | Image Upload | Data Sharing

### DISCUSSION THREAD

**Discussion by Nattyfro** [react-hook-form/react-hook-form#9799] [Q&A] [answered]
Greetings developers,

I'm sure you can solve this issue! I'm currently building a form with two child components. One component allows users to upload an image, and the other should display the image. I've tried using the useWatch.api, but it doesn't seem to have any effect. In comparison, I have no issue making the form work with a simple text field.

(Parent) -  Hompage.tsx

(Child) - AvatarUpload.tsx (uploads the image)
(Child) - AccountCard.tsx (Also Displays the uploaded image)

Typescript
React
React-Hook-Form
useWatch
useEffect
Controller

Thank you in advance for your assistance!

CodeSandBox Link: https://codesandbox.io/s/react-hook-form-usewatch-image-upload-data-sharing-oh7b9n?file=/src/styles.css

Sincerely,
Myles Y

**Comment by Moshyfawn** [comment, marked as answer]
Hey, Myles!

You're forgetting to pass the generated image to RHF state via `onChange`.

Here's a quick draft of how I see the `AvatarUpload` component and [a codesandbox](https://codesandbox.io/s/react-hook-form-usewatch-image-upload-data-sharing-forked-2ojdjb?file=%2Fsrc%2FAvatarUpload.tsx) _(keep in mind that I've omitted most of your business logic for brevity)_

```jsx
import { useCallback } from "react"

import { useController } from "react-hook-form"

function AvatarUpload() {
  const { field } = useController({ name: 'avatar' })
  const { value, onChange } = field

  const handleChange = useCallback(
    event => {
      const file = event.target.files?.[0]
      if (file) {
        generatePreview({ file, onSuccess: (preview) => onChange(preview) })
      }
    },
    [onChange]
  );

  return (
    <div className="Input">
      <img src={value} />
      <input type="file" onChange={handleChange} name="avatar" />
    </div>
  )
}
```

**Reply by Nattyfro** [reply]
Thank you @Moshyfawn, for your assistance! I have taken the necessary precautions to secure my confidential data by deleting the screenshot. If I'm this careless when posting a thread on GitHub, I'm sure I'll make a lot of hackers wealthy once they discover my website! 🤣🤣
